pkg/controller/acicontainersoperator.go

```diff
@@ -481,17 +569,88 @@ func (t *OperatorHandler) ObjectCreated(obj interface{}) {
 			err = rclient.Update(ctx, clusterConfig)
 			if err != nil {
 				log.Info(err)
-				return
+				return true
 			}
 		}
 	}
+
+	log.Info("Adding Aci Operator OwnerRefrence to resources ....")
+	c.indexMutex.Lock()
+	if (!(c.UpdateDeploymentOwnerReference(acicontainersoperator))){
+		log.Info("Error Updating Deployment Owner Reference")
```

Comment:
This will be problem if you return from here without unlock, may be it is better move your lock inside the function so that you no need to take care every where